universe/ignore_records.go

```diff
@@ -91,6 +97,85 @@ func (i *IgnoreTuple) Record() tlv.Record {
 	)
 }
 
+// Encode serializes the IgnoreTuple to the given io.Writer.
+func (i *IgnoreTuple) Encode(w io.Writer) error {
+	stream, err := tlv.NewStream(i.Record())
+	if err != nil {
+		return err
+	}
+
+	return stream.Encode(w)
+}
+
+// Bytes returns the serialized IgnoreTuple.
+func (i *IgnoreTuple) Bytes() ([]byte, error) {
+	var buf bytes.Buffer
+	err := i.Encode(&buf)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+// Digest returns the SHA256 digest of the TLV serialized IgnoreTuple.
+func (i *IgnoreTuple) Digest() ([sha256.Size]byte, error) {
+	var zero [sha256.Size]byte
+
+	ignoreBytes, err := i.Bytes()
+	if err != nil {
+		return zero, err
+	}
+
+	digest := sha256.Sum256(ignoreBytes)
+	return digest, nil
+}
+
+// GenSig generates a Schnorr signature over the IgnoreTuple using the
+// provided SignerClient and key locator.
+func (i *IgnoreTuple) GenSig(ctx context.Context, signer lndclient.SignerClient,
```

Comment:
What about instead having a method that accepts the same params, but returns a [`SIgnedIgnoreTuple`](https://github.com/lightninglabs/taproot-assets/blob/49f92bdcd42ff753cc0747a55e6b33991f8e28fc/universe/ignore_records.go#L218-L225)?